File: 53.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        history_max, current = nums[0], nums[0]
        for item in nums[1:]:
            current = max(item, item+current)
            history_max = max(history_max, current)
        return history_max

File: 53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res = INT_MIN, curSum = 0;
        for (auto num : nums){
            curSum = max(curSum + num, num);
            res = max(curSum, res);
        }
        return res;
    }
};
